server/resp/src/test/java/org/infinispan/server/resp/TransactionOperationsTest.java

```diff
@@ -94,4 +100,83 @@ public void testStartNestedTx() throws Exception {
 
       multi.closeAsync().get(10, TimeUnit.SECONDS);
    }
+
+   @Test
+   public void testTransactionAbortWithError() {
+      RedisCommands<String, String> redis = redisConnection.sync();
+
+      assertThat(redis.multi()).isEqualTo("OK");
+      assertThat(redisConnection.isMulti()).isTrue();
+
+      // This returns an -ERR, but lettuce just returns null when in TX context.
+      assertThat(redis.watch("something")).isNull();
+
+
+      assertThatThrownBy(redis::exec)
+            .isInstanceOf(RedisCommandExecutionException.class)
+            .hasMessage("EXECABORT Transaction discarded because of previous errors.");
+   }
+
+   @Test
+   public void testTransactionWithWatcher() {
+      RedisCommands<String, String> redis = redisConnection.sync();
+
+      assertThat(redis.watch("tx-watcher-k1")).isEqualTo("OK");
```

Comment:
very nitpicking, but should not be OK a constant ?

Reply:
Updated. Imported the test constant.